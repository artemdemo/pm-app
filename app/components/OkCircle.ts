import {Component, Output, EventEmitter, Input} from 'angular2/core';

@Component({
    selector: 'ok-circle',
    directives: [],
    template: `
        <span class="ok-circle-container" (click)="toggleStatus()">
            <span class="ok-circle" [ngClass]="{'ok-circle_done': status}">
                <span class="glyphicon glyphicon-ok-circle" aria-hidden="true"></span>
            </span>
            <span class="ok-circle-content" [ngClass]="{'ok-circle-content_done': status}">
                <ng-content></ng-content> 
            </span>
        </span>
    `
})
export class OkCircle {
    @Input() status: boolean;
    @Output() toggled: EventEmitter<boolean> = new EventEmitter<boolean>();

    constructor() {}

    ngOnInit() {}

    toggleStatus() {
        this.status = !this.status;
        this.toggled.emit(this.status);
    }
}
