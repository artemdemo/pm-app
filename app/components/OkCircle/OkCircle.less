@import '../../styles/colors';

.ok-circle-container {
    cursor: pointer;
}

.ok-circle {
    color: #CACACA;
    font-size: 1.2em;
}
    .ok-circle_done {
        color: @color-success;
    }

.ok-circle-content {
    display: inline-block;
    position: relative;
    top: -2px;
}
    .ok-circle-content_done {
        text-decoration: line-through;
    }
