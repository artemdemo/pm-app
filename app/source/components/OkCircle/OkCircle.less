@import '../../styles/colors';

.ok-circle-container {
    cursor: pointer;
}

.ok-circle {
    color: #CACACA;
    font-size: 1.2em;
}
    .ok-circle_done {
        color: @color-success;
    }
    .ok-circle_with-text {
        margin-right: 5px;
    }

.ok-circle-content {
    display: inline-block;
    position: relative;
    top: -2px;
}
    .ok-circle-content_done {
        text-decoration: line-through;
    }
