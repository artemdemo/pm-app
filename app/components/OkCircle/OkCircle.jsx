import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './OkCircle.less';

export class OkCircle extends Component {
    constructor(props) {
        super(props);

        this.state = {
            doneStatus: props.doneStatus,
        };

        this.renderOkCircle = (loading) => {
            const classOkCircle = classNames({
                'ok-circle': true,
                'ok-circle_with-text': !!this.props.children,
                'ok-circle_done': this.state.doneStatus,
            });
            if (!loading) {
                return (
                    <span className={classOkCircle}>
                        <span
                            className='glyphicon glyphicon-ok-circle'
                            aria-hidden='true'
                        />
                    </span>
                );
            }
            return null;
        };

        this.renderLoadingSpinner = (loading) => {
            if (loading) {
                return (
                    <loading-spinner color='green' />
                );
            }
            return null;
        };

        this.changeDoneStatus = () => {
            const { onChange } = this.props;
            const newDoneStatus = !this.state.doneStatus;
            if (onChange) {
                onChange(newDoneStatus);
            }
        };
    }

    componentWillReceiveProps(nextProps) {
        const { doneStatus } = nextProps;
        this.setState({
            doneStatus,
        });
    }

    render() {
        const classOkCircleContent = classNames({
            'ok-circle-content': true,
            'ok-circle-content_done': this.state.doneStatus,
        });
        const { loading } = this.props;

        return (
            <span
                className='ok-circle-container'
                onClick={this.changeDoneStatus}
            >
                {this.renderOkCircle(loading)}
                {this.renderLoadingSpinner(loading)}
                <span className={classOkCircleContent}>
                    {this.props.children}
                </span>
            </span>
        );
    }
}

OkCircle.propTypes = {
    doneStatus: PropTypes.bool,
    loading: PropTypes.bool,
    onChange: PropTypes.func,
};

OkCircle.defaultProps = {
    doneStatus: false,
    loading: false,
};
