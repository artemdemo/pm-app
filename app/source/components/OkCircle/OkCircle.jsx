import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from '../../components/Icon/Icon';

import './OkCircle.less';

class OkCircle extends React.PureComponent {
    changeDoneStatus = () => {
        const { doneStatus, onChange } = this.props;
        onChange && onChange(!doneStatus);
    };

    render() {
        const { doneStatus } = this.props;
        const classOkCircleContent = classNames({
            'ok-circle-content': true,
            'ok-circle-content_done': doneStatus,
        });
        const classOkCircle = classNames({
            'ok-circle': true,
            'ok-circle_with-text': !!this.props.children,
            'ok-circle_done': doneStatus,
        });

        return (
            <span
                className='ok-circle-container'
                onClick={this.changeDoneStatus}
            >
                <span className={classOkCircle}>
                    <Icon name='check-circle' />
                </span>
                <span className={classOkCircleContent}>
                    {this.props.children}
                </span>
            </span>
        );
    }
}

OkCircle.propTypes = {
    doneStatus: PropTypes.bool,
    onChange: PropTypes.func,
};

OkCircle.defaultProps = {
    doneStatus: false,
    onChange: null,
};

export default OkCircle;
