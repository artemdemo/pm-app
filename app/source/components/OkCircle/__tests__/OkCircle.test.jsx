import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import OkCircle from '../OkCircle';

describe('<OkCircle>', () => {
    it('empty render', () => {
        const tree = renderer.create(
            <OkCircle />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with text', () => {
        const tree = renderer.create(
            <OkCircle >some text</OkCircle>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with done status', () => {
        const tree = renderer.create(
            <OkCircle doneStatus />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should trigger onChange', () => {
        const onChangeMock = jest.fn();
        const wrapper = mount(
            <OkCircle
                onChange={onChangeMock}
            >
                some text
            </OkCircle>
        );
        wrapper.find('.ok-circle-container').simulate('click');
        expect(onChangeMock).toBeCalled();
    });
});
