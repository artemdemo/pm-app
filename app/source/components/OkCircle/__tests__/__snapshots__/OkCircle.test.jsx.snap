// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OkCircle> empty render 1`] = `
<span
  className="ok-circle-container"
  onClick={[Function]}
>
  <span
    className="ok-circle"
  >
    <span
      className="far fa-check-circle"
    />
  </span>
  <span
    className="ok-circle-content"
  />
</span>
`;

exports[`<OkCircle> should render with done status 1`] = `
<span
  className="ok-circle-container"
  onClick={[Function]}
>
  <span
    className="ok-circle ok-circle_done"
  >
    <span
      className="far fa-check-circle"
    />
  </span>
  <span
    className="ok-circle-content ok-circle-content_done"
  />
</span>
`;

exports[`<OkCircle> should render with text 1`] = `
<span
  className="ok-circle-container"
  onClick={[Function]}
>
  <span
    className="ok-circle ok-circle_with-text"
  >
    <span
      className="far fa-check-circle"
    />
  </span>
  <span
    className="ok-circle-content"
  >
    some text
  </span>
</span>
`;
